oteljava/metrics/src/main/scala/org/typelevel/otel4s/oteljava/metrics/CounterBuilderImpl.scala

```diff
@@ -28,12 +28,12 @@ private[oteljava] case class CounterBuilderImpl[F[_]](
     unit: Option[String] = None,
     description: Option[String] = None
 )(implicit F: Sync[F])
-    extends SyncInstrumentBuilder[F, Counter[F, Long]] {
-  type Self = CounterBuilderImpl[F]
+    extends Counter.Builder[F, Long] {
 
-  def withUnit(unit: String): Self = copy(unit = Option(unit))
+  def withUnit(unit: String): Counter.Builder[F, Long] =
+    copy(unit = Option(unit))
```

Comment:
are we deliberately allowing for `null` values of `unit`?

Reply:
`unit` is non-mandatory https://opentelemetry.io/docs/specs/otel/metrics/api/#synchronous-instrument-api. Or am I missing something else?

Reply:
`Option(...)` is for values which may be `null`; `Some(...)` is for values that are assumed not to be `null`. I have complained about this being a bad API before, but the consensus was that it wasn't worth bothering users by deprecating it. `Option.whenNonNull` would be a better API/name than `Option.apply` for that functionality, but alas

Reply:
In this particular case I would prefer silently dropping null values. 

It's much easier than dealing with `Some(null)` in the internals of the SDK backend. 

If we ever implement validation rules (IIRC spec says we shouldn't validate `unit`), we can use:

```scala
require(isValid(unit)) 
Some(unit)
```